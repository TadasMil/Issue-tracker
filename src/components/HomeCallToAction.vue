<script setup lang="ts" name="HomeCallToAction">
import { computed } from 'vue'

const toOpenIssuePage = computed(() => '/issues/open')
</script>

<template>
  <b-card class="shadow-sm border-0 text-center">
    <b-card-title>Having an issue?</b-card-title>

    <b-link
      :to="toOpenIssuePage"
      class="d-flex text-decoration-none text-muted fa-1x justify-content-center align-items-center"
    >
      Submit issue
      <i class="bi bi-arrow-right-short fa-2x"></i>
    </b-link>
  </b-card>
</template>
