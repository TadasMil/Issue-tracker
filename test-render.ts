import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
