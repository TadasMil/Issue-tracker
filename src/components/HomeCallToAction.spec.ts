import { describe, it, expect } from 'vitest'
import { render } from '../../test-render'

import HomeCallToAction from './HomeCallToAction.vue'

describe('HomeCallToAction', () => {
  it('should render the component', () => {
    const { html } = render(HomeCallToAction)

    expect(html()).toMatchSnapshot()
  })
})
