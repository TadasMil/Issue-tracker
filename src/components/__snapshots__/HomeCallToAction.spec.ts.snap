// Vitest Snapshot v1

exports[`HomeCallToAction > should render the component 1`] = `
"<div class=\\"card shadow-sm border-0 text-center\\">
  <!---->
  <!---->
  <div class=\\"card-body\\">
    <!---->
    <!---->
    <h4 class=\\"card-title\\">Having an issue?</h4><a class=\\"d-flex text-decoration-none text-muted fa-1x justify-content-center align-items-center\\" to=\\"/issues/open\\" href=\\"/issues/open\\" target=\\"_self\\"> Submit issue <i class=\\"bi bi-arrow-right-short fa-2x\\"></i></a>
  </div>
  <!---->
  <!---->
</div>"
`;
